src/Azure/Queues/AzureServiceBusQueue.cs

```diff
@@ -133,40 +149,50 @@ public class AzureServiceBusQueue<T> : QueueBase<T> where T : class {
                     if (!queueEntry.IsAbandoned && !queueEntry.IsCompleted)
                         await queueEntry.AbandonAsync().AnyContext();
                 }
+            }, new OnMessageOptions {
+                AutoComplete = false
             });
```

Comment:
What does this control? Shouldn't the auto complete setting be set from the constructor?